### ISSUE TITLE
How to train a pretrained model on image with 6 input channels(2 concatenated RGBs)?

### ISSUE THREAD

**Issue by farnaznouraei** [facebookresearch/detectron2#2062]
Hi. I am trying to train a pretrained model on 6-channel input data (RGB image concatenated with RGB optical flow image) and need to get segmentation masks for 2 classes as output. I looked through other issues (#1292) and noticed the Dataloader needs to be changed but since my data format is still RGB but I have two concatenated images, I was confused how I should change the Dataloader to fit my purpose. I also not sure if there are other things I would need to change for my case and how I should change them (e.g. the MODEL.PIXEL_MEAN , MODEL.PIXEL_STD or INPUT.FORMAT)
Also, would it matter to the model with what package I do the concatenation?

Thanks,

**Comment by ppwwyyxx** [comment]
https://detectron2.readthedocs.io/tutorials/data_loading.html explains how to implement a custom dataloader. It can be implemented to return an image of 6 channels. 
`PIXEL_MEAN/STD` needs to has 6 elements. `INPUT.FORMAT` will not be used if you implement a custom dataloader.

**Comment by farnaznouraei** [comment]
Thanks for your prompt response. In the link you shared, the custom mapper function takes as argument of read_image(), format = 'BGR' and as I looked into this function, PIL is used to read the image which is why it only accepts RGB or HSV formats for this argument. I was wondering what format my 6-channel image falls into. If you could point me to any useful links or examples that use other methods for reading image I would appreciate it.

**Comment by ppwwyyxx** [comment]
You'll need to call a different reading function (e.g. a different library) in the custom dataloader if you need to read a nonstandard format.
